src/eth_rpc/servers/txpool_rpc.rs

```diff
@@ -0,0 +1,50 @@
+use crate::eth_provider::provider::EthereumProvider;
+use crate::eth_rpc::api::txpool_api::TxPoolApiServer;
+use jsonrpsee::core::{async_trait, RpcResult as Result};
+use reth_primitives::Address;
+use reth_rpc_types::txpool::{TxpoolContent, TxpoolContentFrom};
+use tracing::trace;
+
+/// The RPC module for implementing the Txpool api
+#[derive(Debug)]
+pub struct TxpoolRpc<P: EthereumProvider> {
+    eth_provider: P,
+}
+
+impl<P: EthereumProvider> TxpoolRpc<P> {
+    pub const fn new(eth_provider: P) -> Self {
+        Self { eth_provider }
+    }
+
+    async fn content(&self) -> Result<TxpoolContent> {
```

Comment:
Wouldn't it make more sense to expose this in the EthereumProvider trait?

Reply:
I'm not sure for the following two reasons:
- On the one hand I think it is better to keep an architecture quite close to reth which uses a similar mechanism (https://github.com/paradigmxyz/reth/blob/db9c559247dcd2cce2d19decd79b8e3a313c91f3/crates/rpc/rpc/src/ txpool.rs#L33). And I tell myself that for future developments where this function could become more complex, it might be better to keep it in its dedicated environment.
- I wanted to avoid further overloading the `EthereumProvider` with additional logic and functions. I put the bare minimum there to just get the content of the mem pool and nothing else.

What do you think?

Reply:
I agree with the Reth implementation but the difference is that Reth implements `content` on the Pool itself, whereas here we are implementing it on the RPC which looks a little weird. 
I would personally have the little logic you added in content just moved to the `EthereumProvider`, which for now agglomerates all the `ETH` logic.

Also once we implement the mempool, we will most like the pool be part of the `EthDataProvider`, which will call `content` on the `TxPool` and return the `TxpoolContent`.

Reply:
Ok understood, just migrated this